azkaban-web-server/src/main/java/azkaban/imagemgmt/daos/ImageVersionDaoImpl.java

```diff
@@ -0,0 +1,132 @@
+package azkaban.imagemgmt.daos;
+
+import azkaban.db.DatabaseOperator;
+import azkaban.db.SQLTransaction;
+import azkaban.imagemgmt.exeception.ImageMgmtException;
+import azkaban.imagemgmt.models.ImageType;
+import azkaban.imagemgmt.models.ImageVersion;
+import azkaban.imagemgmt.models.ImageVersion.State;
+import azkaban.imagemgmt.models.RequestContext;
+import azkaban.imagemgmt.utils.ImageMgmtConstants;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class ImageVersionDaoImpl implements ImageVersionDao {
+
+  private static final Logger log = LoggerFactory.getLogger(ImageVersionDaoImpl.class);
+
+  private final DatabaseOperator databaseOperator;
+
+  private final ImageTypeDao imageTypeDao;
+
+  private static String INSERT_IMAGE_VERSION_QUERY =
+      "insert into image_versions ( path, description, version, type_id, state, "
+          + "release_tag, created_on, created_by) "
```

Comment:
no need to pass created_on. Default should take care of it. For modified_on, can you please see if you can add default as well as update clause? Something like "modified_on DATETIME DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP"

Reply:
Done.